apps/retrospective/services/sprint_sync.go

```diff
@@ -336,29 +340,74 @@ func (service SprintService) AssignPoints(sprintID string, sprintTaskID *string)
 	return nil
 }
 
+// GetTimeTrackerData ...
+func (service SprintService) GetTimeTrackerData(sprint retroModels.Sprint, taskProviderConfig []byte) (
+	[]string,
+	map[uint][]timeTrackerSerializers.TimeLog,
+	error) {
+
+	var timeTrackerTaskKeys []string
+	var timeLogs []timeTrackerSerializers.TimeLog
+	sprintMemberTimeLogs := map[uint][]timeTrackerSerializers.TimeLog{}
+	var err error
+
+	if sprint.Retrospective.TimeProviderName == constants.JIRA {
+		timeTrackerTaskKeys, timeLogs, err = service.GetSprintMemberTimeTrackerData(taskProviderConfig, sprint)
+		if err != nil {
+			utils.LogToSentry(err)
+			service.SetSyncFailed(sprint.ID)
+			return nil, nil, err
+		}
+		memberEmailTimeLogs := make(map[string][]timeTrackerSerializers.TimeLog)
+		for _, timeLog := range timeLogs {
+			memberEmailTimeLogs[timeLog.Email] = append(memberEmailTimeLogs[timeLog.Email], timeLog)
+		}
+		for _, sprintMember := range sprint.SprintMembers {
+			sprintMemberTimeLogs[sprintMember.ID] = memberEmailTimeLogs[sprintMember.Member.Email]
+		}
+	} else {
+		for _, sprintMember := range sprint.SprintMembers {
+			var memberTaskKeys []string
+			memberTaskKeys, timeLogs, err = service.GetSprintMemberTimeTrackerData(sprintMember.Member.TimeProviderConfig, sprint)
+			if err != nil {
+				utils.LogToSentry(err)
+				service.SetSyncFailed(sprint.ID)
+				return nil, nil, err
+			}
+			sprintMemberTimeLogs[sprintMember.ID] = timeLogs
+			timeTrackerTaskKeys = append(timeTrackerTaskKeys, memberTaskKeys...)
+		}
+	}
+
+	return timeTrackerTaskKeys, sprintMemberTimeLogs, nil
+}
+
 // GetSprintMemberTimeTrackerData ...
 func (service SprintService) GetSprintMemberTimeTrackerData(
-	sprintMember retroModels.SprintMember,
+	timeTrackerConfig []byte,
 	sprint retroModels.Sprint) ([]string, []timeTrackerSerializers.TimeLog, error) {
 
 	timeLogs, err := timetracker.GetProjectTimeLogs(
-		sprintMember.Member.TimeProviderConfig,
+		timeTrackerConfig,
 		sprint.Retrospective.ProjectName,
 		*sprint.StartDate,
 		*sprint.EndDate)
 
 	if err != nil {
 		utils.LogToSentry(err)
+		service.SetSyncFailed(sprint.ID)
 		return nil, nil, err
 	}
-
+	sprintMemberEmailMap := make(map[string]uint)
+	for _, member := range sprint.SprintMembers {
+		sprintMemberEmailMap[member.Member.Email] = member.MemberID
+	}
 	var ticketKeys []string
 	for _, timeLog := range timeLogs {
-		ticketKeys = append(ticketKeys, timeLog.TaskKey)
-		if err != nil {
-			utils.LogToSentry(err)
-			return nil, nil, err
+		if _, exists := sprintMemberEmailMap[timeLog.Email]; !exists {
+			continue
```

Comment:
Please try to avoid break and continue statements.

Reply:
removed.